src/libcore/marker.rs

```diff
@@ -351,7 +351,35 @@ pub trait PhantomFn<A:?Sized,R:?Sized=()> { }
 /// instance, it will behave *as if* an instance of the type `T` were
 /// present for the purpose of various automatic analyses.
 ///
-/// For example, embedding a `PhantomData<T>` will inform the compiler
+/// # Example
+/// When handling external resources over a foreign function interface,
+/// PhantomData can prevent mismatches by enforcing types in
+/// the method implementations, although the struct doesn't actually
+/// contain values of the resource type.
+///
+///```
+///pub struct ExternalResource<R> {
```

Comment:
Does this pass the tests? I thought (though I could be wrong) that using `pub` like this causes a test failure.